core/src/main/java/hudson/FilePath.java

```diff
@@ -2655,13 +2655,16 @@ private void readFromTar(String name, File baseDir, InputStream in) throws IOExc
                         if (mode != 0 && !Functions.isWindows()) // be defensive
                             _chmod(f, mode);
                     }
+                    files.add(f);
                 }
             }
         } catch (IOException e) {
             throw new IOException("Failed to extract " + name, e);
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt(); // process this later
             throw new IOException("Failed to extract " + name, e);
+        }finally {
+            return files;
```

Comment:
all exception will be ignored. I would not do that. and in case of any  exception you will return an empty list without notifying a problem happened to the caller

Reply:
would be nice to return a custom exception with all successfully untar'd files.

Reply:
Unnecessary if you switch to a callback model, and let the caller print files as they are processed.

Reply:
FWIW, I see callbacks quite confusing, and mixing responsibilities. I'd rather avoid it.